pyrobosim/pyrobosim/core/world.py

```diff
@@ -683,13 +687,18 @@ def check_occupancy(self, pose):
         :return: True if the pose is occupied, else False if free.
         :rtype: bool
         """
+        if self.occupancy_grid:
```

Comment:
Maybe worth additionally having `check_occupancy()` take an optional argument `use_occupancy_grid`, so that even if there is a grid, we can always choose to continue using polygons for checking as needed.

Reply:
yes, 
we can propagate  the `use_grid`  from the planner .
Would there be a case that a planner was initialized with `use_grid=True`  and then the same planner will need switch from using grid to polygons ?

Reply:
I don't think so. If anything, a new planner can be instantiated.